src/components/common/ImgSlide/ImgSlide.js

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+import * as S from './Img.style';
+
+const ImgSlide = ({ list }) => {
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [newImgArr, setNewImgArr] = useState([]);
+  const slideImageBox = useRef();
+
+  const settingArr = [...list];
+
+  const nextIndex = currentIndex === list.length - 1 ? 0 : currentIndex + 1;
+  const prevIndex = currentIndex === 0 ? list.length - 2 : currentIndex - 2;
+
+  useEffect(() => {
+    settingArr.push(list[0]);
+    settingArr.unshift(list[list.length - 1]);
+    setNewImgArr(settingArr);
+  }, []);
```

Comment:
```suggestion
  const nextIndex = currentIndex === list.length - 1 ? 0 : currentIndex + 1;
  const prevIndex = currentIndex === 0 ? list.length - 2 : currentIndex - 2;

  useEffect(() => {
    const settingArr = [...list];
    settingArr.push(list[0]);
    settingArr.unshift(list[list.length - 1]);
    setNewImgArr(settingArr);
  }, []);
```
- useEffect 최초 실행될 때만 사용될 변수가 함수 body에 있으니 헷갈리네요. 같이 쓰이는 값들은 한 곳에 모아놓는게 좋겠습니다
- 사실 애초에 이런 연산들은 useState 초기값에 넣어주면 useEffect를 사용하지 않고도 초기값으로 넣을 수 있습니다